### ISSUE TITLE
Cosmic becomes unresponsive using Steam

### ISSUE THREAD

**Issue by julianbraha** [pop-os/cosmic-epoch#1144]
### The bug:
Every time I hover over my username in Steam, my system freezes and becomes completely unresponsive (e.g. cursor doesn't move, keyboard shortcuts don't do anything).

### Debugging info:
This doesn't happen in KDE Plasma 6.2.1, so I think it's specific to Cosmic.

I tried running `sudo dmesg --follow` in a terminal, as well as the `bottom` resource monitor in another terminal while reproducing this error, but `dmesg` doesn't show any useful errors, and the resource monitor doesn't show any spikes (**see screenshots below**).

Notice that the dropdown menu that should open when hovering over the username, doesn't render correctly, and is an opaque black box.

### My system:
Hardware
- Framework Laptop 13
- AMD Ryzen 5 7640U
- 32gb DDR5 @ 4800 MT/s, CL34
- SK hynix Platinum P41 2TB

Software
- Operating system: CachyOS
- Kernel: 6.11.3-3-cachyos
- Graphics driver in use: `amdgpu`

### Photos, since the system is too unresponsive for screenshots to work:
![PXL_20241017_122225338](https://github.com/user-attachments/assets/5315a23b-527b-4255-a3aa-e4c0d530b512)
![PXL_20241017_122503331](https://github.com/user-attachments/assets/5ef211f6-64b7-43aa-93bb-a5918eacab19)

**Comment by julianbraha** [comment]
Update on this:

When I open Steam by running in a terminal (Alacritty): `cosmic-comp steam`, the issue doesn't happen at all. The username dropdown renders as expected, and there is no system freeze.

But when I open Steam by clicking on the desktop icon, or running in terminal `steam`, this issue continues.

**Comment by jokeyrhyme** [comment]
I've been running Steam-in-flatpak ( https://flathub.org/apps/com.valvesoftware.Steam ) for months now, also on an AMD GPU, and haven't run into this issue

I wonder if there's a difference between the flatpak version and the CachyOS-distributed package that is significant here?

**Comment by julianbraha** [comment]
> I've been running Steam-in-flatpak ( https://flathub.org/apps/com.valvesoftware.Steam ) for months now, also on an AMD GPU, and haven't run into this issue
> 
> I wonder if there's a difference between the flatpak version and the CachyOS-distributed package that is significant here?

No, I tried installing and running the Steam flatpak, and still run into this issue, where dropdowns don't render, and cause the entire system to hang.

See pic:
![PXL_20241110_133116861](https://github.com/user-attachments/assets/3eb886a9-beb1-4529-a502-75cf181ec98a)

**Comment by jokeyrhyme** [comment]
Oh, which vulkan driver are you using: AMDVLK or RADV? I uninstalled AMDVLK because it was breaking a game, and this was in GNOME

**Comment by Arthur-GYT** [comment]
> Oh, which vulkan driver are you using: AMDVLK or RADV? I uninstalled AMDVLK because it was breaking a game, and this was in GNOME

Can you explain how to use radv and uninstall amdvlk?

**Comment by jokeyrhyme** [comment]
https://wiki.archlinux.org/title/Vulkan#Installation

https://wiki.archlinux.org/title/Vulkan#Switching_between_AMD_drivers

**Comment by julianbraha** [comment]
> Oh, which vulkan driver are you using: AMDVLK or RADV? I uninstalled AMDVLK because it was breaking a game, and this was in GNOME

I was using AMDVLK (closed version), but I've uninstalled this and tried with RADV. 

Still getting the issue :(

**Comment by Arthur-GYT** [comment]
Seems to work with the latest update

**Comment by Arthur-GYT** [comment]
Please add this line in your `.bashrc` file and restart your computer

```bash
export COSMIC_DISABLE_DIRECT_SCANOUT="1"
```

**Comment by julianbraha** [comment]
> Please add this line in your `.bashrc` file and restart your computer
> 
> ```shell
> export COSMIC_DISABLE_DIRECT_SCANOUT="1"
> ```

I've added this line to my `.bashrc` but I'm still getting the issue :(

I'm using the latest release packaged by CachyOS.

**Comment by julianbraha** [comment]
I reinstalled CachyOS with the Cosmic desktop, and still got this issue. So I tried following all of the suggestions above, again.

The following fixed it:
1. `export COSMIC_DISABLE_DIRECT_SCANOUT="1"` (actually I had to set it in my `config.fish` like `set -x COSMIC_DISABLE_DIRECT_SCANOUT 1`)
2. uninstalling `amdvlk`, and using `vulkan-radeon`

(using flatpak was unnecessary)

Thanks everyone. Closing.
